kubernetes-addons/irsa/main.tf

```diff
@@ -31,7 +32,7 @@ resource "kubernetes_namespace" "add_on_ns" {
 resource "kubernetes_service_account" "add_on_sa" {
   metadata {
     name        = var.kubernetes_service_account
-    namespace   = kubernetes_namespace.add_on_ns.id
+    namespace   = var.kubernetes_namespace
```

Comment:
This might cause an issue for new namespaces as this has no dependency on `resource "kubernetes_namespace" "add_on_ns"` 
Should we use local variable that validates? something like

```hcl
k8s_namespace = lookup(data.kubernetes_namespace.namespace, "id") == null ? kubernetes_namespace.add_on_ns[0].id: var.kubernetes_namespace
```

Worth testing both scenarios

Reply:
yeah that makes sense, added `lookup()` to `locals.tf` in irsa.